### PR TITLE
Feature/565 dropdown component

### PR DESCRIPTION
**Pull request by coltraneyan**
- New reusable dropdown component
- Styled according to this version below from Figma
178142351-ee911c42-8dd5-4dd6-9447-f2fa05f02a66

- in addition to options and current values, it also recieves a width prop to create a static width the parent can adjust according to the string length of the options
- all issues should be resolved from [this draft PR](https://github.com/airswap/airswap-web/pull/577) @piersss

### CODE REVIEW

File: src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          itemWidth={width}
+          onClick={() => {
+            handleShow();
+          }}
+        >
+          {options.map((option, index) => {
+            return (
+              <Item
+                isSelected={
+                  value
```
Comment (GiuntaLucas) [inline src/components/Dropdown/Dropdown.tsx:57]: You can optimise and just write value ? value.label === option.label : index === 0. Don't realy need to return true or false, the condition will do the job ^^

Comment (coltraneyan) [reply src/components/Dropdown/Dropdown.tsx:57]: Nice catch, thanks.

File: src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import Icon from "../Icon/Icon";
+import { Item, Wrapper, AbsoluteWrapper, Current } from "./Dropdown.styles";
+
+export type SelectOption = {
+  label: string;
+  value: string;
+};
+
+export type DropdownProps = {
+  value?: SelectOption;
```
Comment (piersss) [inline src/components/Dropdown/Dropdown.tsx:12]:
```suggestion
  value: SelectOption;
```

In the designs there's always a default state. So `value` should not be optional.

File: src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  width?: number;
+};
+
+export const Dropdown: React.FC<DropdownProps> = ({
+  value,
+  options,
+  onChange,
+  width,
+}) => {
+  const [showOptions, setShowOptions] = useState<Boolean>(false);
```
Comment (piersss) [inline src/components/Dropdown/Dropdown.tsx:24]:
```suggestion
```

We're using `focus` so this is also not needed.

File: src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                  handleChange(option);
+                }}
+              >
+                {option.label}
+              </Item>
+            );
+          })}
+        </AbsoluteWrapper>
+      )}
+    </Wrapper>
```
Comment (piersss) [inline src/components/Dropdown/Dropdown.tsx:75]:
```suggestion
return (
    <Wrapper className={className}>
      <Current>
        {value.label}
        <Icon name={"chevron-down"} />
      </Current>
      <AbsoluteWrapper>
        {options
          .filter(option => option.value !== value.value)
          .map((option, index) => {
          return (
            <Item
              key={option.value}
              onClick={() => {
                handleChange(option);
              }}
            >
              {option.label}
            </Item>
          );
        })}
      </AbsoluteWrapper>
    </Wrapper>
```

In the end you just need a lot less. Added a filter to filter out the active option.

File: src/components/Dropdown/Dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  label: string;
+  value: string;
+};
+
+export type DropdownProps = {
+  value?: SelectOption;
+  options: SelectOption[];
+  onChange: (option: SelectOption) => void;
+  width?: number;
+};
```
Comment (piersss) [inline src/components/Dropdown/Dropdown.tsx:16]:
```suggestion
  className?: string;
};
```

Styled-components needs `className` to let a parent style it's children.

https://styled-components.com/docs/basics#styling-any-component

**Review by piersss — APPROVED** [review]

**Review by GiuntaLucas — APPROVED** [review]

**Pull request merged**